### PR TITLE
Improve runtests

### PR DESCRIPTION
**Pull request by luksa**
- split results dir into separate, per-version dirs
- store entire log output to output.log
- store logs of failed test to failed.log
- store skipped tests to skipped.log
- include "DONE" and the next line (such as "ERROR ending test run because max failures was reached") in test summary

### CODE REVIEW

**Comment by luksa** [comment]
Please review only the last commit ("Improve runtests.sh"). The other two are in #506 and #507.

/hold until #506 and #507 are merged

**Review by fjglira — APPROVED** [review]

**Pull request merged**
